src/ragas/metrics/context_relevance.py

```diff
@@ -2,62 +2,137 @@
+Now given a question and context, extract the minimum number of sentences from the given context required to answer the question completely. 
+sentences:Scott Derrickson (born July 16, 1966) is an American director, screenwriter and producer. Edward Davis Wood Jr. (October 10, 1924 – December 10, 1978) was an American filmmaker, actor, writer, producer, and director.
+
+question:{}
+context:\n{}
+sentences:"""  # noqa: E501
+
+
+def sent_tokenize(sent: str) -> List[str]:
+    return [s[:-1] if s.endswith(".") else s for s in sent.strip().split(". ")]
+
+
+class SentenceAgreement:
+    def __init__(
+        self: t.Self,
+        model_name: str = "cross-encoder/stsb-TinyBERT-L-4",
+        metric: str = "bert_score",
+    ):
+        self.metric = metric
+        self.cross_encoder = CrossEncoder(model_name)
+
+    def bert_score(self, para1: str, para2: str) -> float:
+        sentences1, sentences2 = sent_tokenize(para1), sent_tokenize(para2)
+        scores = self.cross_encoder.predict(
+            list(product(sentences1, sentences2)), convert_to_numpy=True  # type: ignore
+        )
+        assert isinstance(scores, np.ndarray), "Expects ndarray"
+        scores = scores.reshape(len(sentences1), len(sentences2))
+        return scores.max(axis=1).mean()
+
+    @staticmethod
+    def jaccard_score(para1: str, para2: str) -> float:
+        sentences1, sentences2 = sent_tokenize(para1), sent_tokenize(para2)
+        intersect = len(np.intersect1d(sentences1, sentences2))
+        union = len(np.union1d(sentences1, sentences2))
+        return intersect / union
+
+    def evaluate(self, answers: List[List[str]]) -> float:
+        """
+        eval nC2 combinations
+        """
+        scores = []
+        groups = combinations(answers, 2)
+        for group in groups:
+            if self.metric == "jaccard":
+                score = self.jaccard_score(*group)  # type: ignore
+            elif self.metric == "bert_score":
+                score = self.bert_score(*group)  # type: ignore
+            else:
+                score = 0
+                raise ValueError(f"Metric {self.metric} unavailable")
+            scores.append(score)
+        scores = np.mean(scores)
+        assert isinstance(scores, float), "Expects floats"
+        return scores
 
 
 @dataclass
 class ContextRelevancy(Metric):
-    batch_size: int = 32
     name: str = "context_relavency"
-    model_name: str = "t5-base"
+    batch_size: int = 15
+    agreement_metric: str = "bert_score"
+    strictness: int = 2
```

Comment:
what does the strictness param do? would be great to have doc string at this point I guess